crates/evm/src/fuzz/invariant/executor.rs

```diff
@@ -323,26 +319,59 @@ impl<'a> InvariantExecutor<'a> {
     ) -> eyre::Result<()> {
         // targetArtifactSelectors -> (string, bytes4[])[].
         let targeted_abi = self
-            .get_list::<(String, Vec<FixedBytes>)>(
+            .get_list::<(String, Vec<FixedBytes<4>>)>(
                 invariant_address,
                 abi,
                 "targetArtifactSelectors",
+                |v| {
+                    if let Some(list) = v.as_array() {
+                        list.iter().map(|val| {
+                            if let Some((_, _str, elements)) = val.as_custom_struct() {
+                                let name = elements[0].as_str().unwrap().to_string();
+                                let selectors = elements[1]
+                                    .as_array()
+                                    .unwrap()
+                                    .iter()
+                                    .map(|selector| {
+                                        FixedBytes::<4>::from_slice(&selector.as_fixed_bytes().unwrap().0[0..4])
+                                    })
+                                    .collect::<Vec<_>>();
+                                (name, selectors)
```

Comment:
yeah this needs to go we cannot work with an API that is so low level IMO, these type conversions should never surface to the user